specutils/io/default_loaders/tabular_fits.py

```diff
@@ -0,0 +1,96 @@
+import logging
+import os
+
+import six
+from astropy.io import fits
+from astropy.nddata import StdDevUncertainty
+from astropy.table import Table
+import astropy.units as u
+import numpy as np
+from astropy.wcs import WCS
+
+from ...spectra import Spectrum1D
+from ..registers import data_loader, custom_writer
+from ..generic_spectrum_from_table import generic_spectrum_from_table
+
+__all__ = ['tabular_fits_loader', 'tabular_fits_writer']
+
+
+def identify_tabular_fits(origin, *args, **kwargs):
+    # check if file can be opened with this reader
+    # args[0] = filename
+    # fits.open(args[0]) = hdulist
+    return (isinstance(args[0], six.string_types) and
+            # check if file is .fits
+            os.path.splitext(args[0].lower())[1] == '.fits' and
+            # check hdulist has more than one extension
+            len(fits.open(args[0])) > 1 and
+            # check if fits has BinTable extension
+            isinstance(fits.open(args[0])[1], fits.BinTableHDU)
+            )
+
+
+@data_loader("tabular-fits", identifier=identify_tabular_fits,
+             dtype=Spectrum1D, extensions=['fits'])
+def tabular_fits_loader(file_name, column_mapping=None, **kwargs):
+    """
+    Load spectrum from a FITS file.
+
+    Parameters
+    ----------
+    file_name: str
+        The path to the FITS file
+    column_mapping : dict
+        A dictionary describing the relation between the FITS file columns
+        and the arguments of the `Spectrum1D` class, along with unit
+        information. The dictionary keys should be the FITS file column names
```

Comment:
fits tables sometimes have units.  As with ECSV the way this should work is allow `{'FLUX': 'flux'}` to mean "use whatever units are present".  I would also argue that `{'FLUX': ('flux', someunit)}` should either just fail if untis are in the table, or try to auto-convert

Reply:
Okay. Currently, if the table contains unit information for a column, that unit information is used over anything defined in the column mapping. We can extend to auto converting in the future, but we'd have to parse physical type information so that we can pass in the correct equivalencies, so I think the fall back is fine.

Reply:
Wait, so if the user gives a specific unit but the table conflicts it silently does what the table says? That's a classic UX mistake.  Can we maybe just raise an exception in that case and plan to do the better thing later as you say?

Reply:
We can do it the opposite way: prefer the column mapping over the table, but from your first comment

>  I would also argue that {'FLUX': ('flux', someunit)} should either just fail if untis are in the table

sounded like you preferred that the table take precedence over the column mapping. We can raise an exception if _both_ the table and the column mapping define a unit, or log something above preferring one over the other.

Reply:
Physical type information parsing has been added. If a user has defined a unit in the column mapping _and_ a unit is defined in the table already, specutils will attempt to convert the quantity array from the table to the unit defined in the column mapping.

Reply:
I meant want I *really* wanted is something like this: `spec1dkwargs[mapping_value[0]] = table[mapping_key].to(mapping_value[1], whatevertherightequivalency)`.  That is, if the table has units, assume the file knows what its units are, but that the user is asking to *convert* them to whatever the explicitly asked for in `Spectrum1D`.  And if the value is just `'flux'`, trust that the table is correct.

But I'm fine with just raising an exception for now if figuring out `whatevertherightequivalency` is going to be too hard to do right now.

Reply:
I think I agree with @eteq on the desired behavior, and that it's okay to just raise an exception for this release. 

A related question: Are we checking for and/or enforcing consistency between the units for the flux and the uncertainty? If nothing for this is coded yet, I'd suggest issuing a warning if they are in different units.

Reply:
@eteq Yes, this is what's already implemented. Although, I don't think the form of the column mapping dictionary should change if no unit is desired. I.e. it should be `{'FLUX': ('flux', None)}`.

Reply:
@nmearl If someone were to feed flux in erg cm-2 s-1 AA-1 and uncertainties in Jy would calculations like SNR currently give the right answer? If not, perhaps the check should go into Spectrum1D itself, when the object is created?

Reply:
@hcferguson I think the best solution would be to be sure that all statistical functions are performed within an equivalency context manager. I don't know that a user who explicitly wants their flux and error in different units would appreciate either being told no, or having one or the other converted implicitly. We should just do the conversions at calculation time, for the same of the calculation, and leave the Spectrum1D in whatever state the user wants.

Reply:
@nmearl Agreed. But what is it doing now?

Reply:
I think this is a feature not a bug - that the `analysis` functions (which are *not* all statistical), don't make any strong assumptions nor try to guess equivalencies.  I think that's desirable because in *most* cases it does the right thing - e.g. equivalent widths end up with `spectral_axis` units *regardless* of the `flux` units.  

But this is pretty far-afield of this PR, so I'd suggest the discussion be moved to some other issue if any changes need to happen.

Reply:
And I agree that this seems to do exactly what I said I wanted in https://github.com/astropy/specutils/pull/376#discussion_r233978718.  Not sure why I thought otherwise (perhaps I had an out-of-date view when I did part of this review?  Would explain some of the other rather confusing comments...)